### ISSUE TITLE
Issues with extracting field information along AMRVAC streamline

### ISSUE THREAD

**Issue by Kiradorn** [yt-project/yt#3286]
### Bug report

**Bug summary**

I am facing implementation errors when trying to extract field information along a defined streamline. The dataset that is used is from AMRVAC.

**Code for reproduction**

Curldrop --> http://use.yt/upload/bf719e00

Field 't' error:
```python
import yt
import matplotlib.pylab as pl
from yt.visualization.api import Streamlines

overrides = dict(length_unit=(1e8, 'cm'), temperature_unit=(1e6, 'K'), numberdensity_unit=(1e8,'cm**-3'))
file_index=746
ds=yt.load('/Volumes/TRANSCEND/Projects/amrvac/prom_form_3D_10G_linde/jk20210'+str(file_index)+'.dat', units_override=overrides)
ds.force_periodicity()
streamlines = Streamlines(ds, ds.domain_center, 'b1t', 'b2t', 'b3t', length=40.0)
streamlines.integrate_through_volume()
stream = streamlines.path(0)
pl.semilogy(stream['t'], stream['density'], '-x')
```

Implementation error:
```python
print(stream['density])
```

**Actual outcome**

Field 't' error:
![Screenshot 2021-05-20 at 09 39 44](https://user-images.githubusercontent.com/13555304/118938837-5d12cc00-b94f-11eb-82dc-6f5747649f9b.png)

Implementation error:
![Screenshot 2021-05-20 at 09 29 15](https://user-images.githubusercontent.com/13555304/118937538-ee813e80-b94d-11eb-8874-1f6150ab0fb9.png)

**Expected outcome**

According to https://yt-project.org/doc/visualizing/streamlines.html#data-access-along-the-streamline this method should work to extract the field information along the streamline specified.

The 'stream' variable does indeed contain the relevant information, or as best as I can tell:
![Screenshot 2021-05-20 at 09 34 55](https://user-images.githubusercontent.com/13555304/118938244-b9c1b700-b94e-11eb-81c4-4c1ec2e836ca.png)

**Version Information**

  * Operating System: macOS Big Sur 11.2.3
  * Python Version: 3.8.5
  * yt version: 4.0
  * Other Libraries (if applicable):
  Matplotlib

Yt was installed as a clone from github, and setup following 'python setup.py develop'

**Comment by neutrinoceros** [comment]
Notes: although this is showing with an AMRVAC dataset, I'm not sure the issue is frontend specific. And in any case, this is not something I am familiar with so I won't be able to provide help quickly here.

Regarding the `"t"` field, I'm note sure when it's supposed to be generated, and the name isn't very explicit. It's possible the documentation is out of sync here.

**Comment by Kiradorn** [comment]
According to the "Why are the values in my Ray object out of order?" subheading of https://yt-project.org/doc/faq/index.html?highlight=ds.r ```"t"``` is the index along the ray, or in this case the streamline.

**Comment by chummels** [comment]
Correct, `t` is defined for `Ray` objects to indicate your location along the 1D Ray (in unitary units), but I didn't know they were created for streamline objects.

**Comment by Kiradorn** [comment]
Okay so I found a workaround for this last week:

```python
offset=-1 # Sometimes there's an issues with the final point being outside of simulation box so I ignore this position. Might be only amrvac that suffers from this because of lack of ghostcells.
path_num=0 # Chosen fieldline index

#Positions of points from streamline
xarray=streamlines.path(path_num).positions.value[:,0]
filtered_x=xarray[abs(xarray)>0]
yarray=streamlines.path(path_num).positions.value[:,2]
filtered_y=yarray[abs(xarray)>0]

# Only extract those positions that have non-zero positioning - returned if the length of the streamline integration is longer than the dimensions of the box.
filtered_locations=streamlines.path(path_num).positions.value[abs(xarray)>0]

#Plot the field values at those points.
plt.plot(filtered_y[:offset],ds.find_field_values_at_points(('gas','temperature'),filtered_locations[:offset,:]))
```

A slightly more involved code then returns something like the following:
![field_tracing_params_test](https://user-images.githubusercontent.com/13555304/120155233-e39d9800-c1f0-11eb-9910-9a0421bdb39d.png)
